controllers/skipjob/job.go

```diff
@@ -0,0 +1,387 @@
+		StartingDeadlineSeconds: skipJob.Spec.Cron.StartingDeadlineSeconds,
+		ConcurrencyPolicy:       skipJob.Spec.Cron.ConcurrencyPolicy,
+		Suspend:                 skipJob.Spec.Cron.Suspend,
+		JobTemplate: batchv1.JobTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: GetJobLabels(skipJob, jobName, podLabels),
+			},
+			Spec: getJobSpec(skipJob, selector, podLabels, gcpIdentityConfigMap),
+		},
+		SuccessfulJobsHistoryLimit: util.PointTo(int32(3)),
+		FailedJobsHistoryLimit:     util.PointTo(int32(1)),
+	}
+}
+
+func GetJobLabels(skipJob *skiperatorv1alpha1.SKIPJob, jobName string, labels map[string]string) map[string]string {
+	if len(labels) == 0 {
+		labels = make(map[string]string)
+	}
+	labels["job-name"] = jobName
+	labels[SKIPJobReferenceLabelKey] = skipJob.Name
+
+	return labels
+}
+
+func getJobSpec(skipJob *skiperatorv1alpha1.SKIPJob, selector *metav1.LabelSelector, podLabels map[string]string, gcpIdentityConfigMap *corev1.ConfigMap) batchv1.JobSpec {
+	podVolumes, containerVolumeMounts := core.GetContainerVolumeMountsAndPodVolumes(skipJob.Spec.Container.FilesFrom)
+
+	if skipJob.Spec.Container.GCP != nil {
+		gcpPodVolume := gcp.GetGCPContainerVolume(gcpIdentityConfigMap.Data["workloadIdentityPool"], skipJob.Name)
+		gcpContainerVolumeMount := gcp.GetGCPContainerVolumeMount()
+		gcpEnvVar := gcp.GetGCPEnvVar()
+
+		podVolumes = append(podVolumes, gcpPodVolume)
+		containerVolumeMounts = append(containerVolumeMounts, gcpContainerVolumeMount)
+		skipJob.Spec.Container.Env = append(skipJob.Spec.Container.Env, gcpEnvVar)
+	}
+
+	jobSpec := batchv1.JobSpec{
+		Parallelism:           util.PointTo(int32(1)),
+		Completions:           util.PointTo(int32(1)),
+		ActiveDeadlineSeconds: skipJob.Spec.Job.ActiveDeadlineSeconds,
+		PodFailurePolicy:      nil,
+		BackoffLimit:          skipJob.Spec.Job.BackoffLimit,
+		Selector:              nil,
+		ManualSelector:        nil,
+		Template: corev1.PodTemplateSpec{
+			Spec: core.CreatePodSpec(core.CreateJobContainer(skipJob, containerVolumeMounts), podVolumes, skipJob.HashedName(), skipJob.Spec.Container.Priority, skipJob.Spec.Container.RestartPolicy),
+		},
+		TTLSecondsAfterFinished: skipJob.Spec.Job.TTLSecondsAfterFinished,
+		CompletionMode:          util.PointTo(batchv1.NonIndexedCompletion),
+		Suspend:                 skipJob.Spec.Job.Suspend,
+	}
+
+	// Jobs create their own selector with a random UUID. Upon creation of the Job we do not know this beforehand.
+	// Therefore, simply set these again if they already exist, which would be the case if reconciling an existing job.
+	if selector != nil {
+		jobSpec.Selector = selector
+		if jobSpec.Template.ObjectMeta.Labels == nil {
+			jobSpec.Template.ObjectMeta.Labels = map[string]string{}
+		}
+		maps.Copy(jobSpec.Template.ObjectMeta.Labels, podLabels)
+	}
+
+	return jobSpec
+}
+
+func generateDebugContainer(pod *corev1.Pod) (*corev1.Pod, *corev1.EphemeralContainer, error) {
+	ec := &corev1.EphemeralContainer{
```

Comment:
Må det noen spesielle tilganger til for å spinne opp ephemeral containers?
Jeg tenker både for at Skiperator skal få lov og for at OPA skal tillate det i clusteret?

Reply:
Yea, you need to be able to patch the subresource pods/ephemeralcontainers.

Its in controller.go for SKIPJobs:
```
//+kubebuilder:rbac:groups=core,resources=pods;pods/ephemeralcontainers,verbs=get;list;watch;create;update;patch;delete
```

Reply:
AFAIK OPA allows it. I have been able to make it work in Sandbox. I have however not checked out for warnings from gatekeeper, should probably do so